modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -2120,6 +2130,8 @@ private UpdateResult executeUpdate(
                 break;
         }
 
+        heavyQueriesTracker().stopTracking(dmlInfo, null);
```

Comment:
try/finally block? Perhaps in executeDml method instead of executeUpdate?

Reply:
Tracking moved to `IgniteH2Indexing#executeDml`